### PR TITLE
PUT /api/responses

### PR DESCRIPTION
**Pull request by HanzoDaBoss**
Added endpoint
Added testing
Added error handling

### CODE REVIEW

File: back-end/__tests__/app.test.js [hunk 1]
```diff
@@ -491,3 +491,71 @@ describe("/api/teams", () => {
       });
   });
 });
+
+describe.only("/api/responses", () => {
```
Comment (miors) [inline back-end/__tests__/app.test.js]: Need to remove .only

File: back-end/__tests__/app.test.js [hunk 2]
```diff
@@ -491,3 +491,71 @@ describe("/api/teams", () => {
...
+        const { response } = body;
+        expect(response.response_id).toBe(9);
+        expect(response.user_id).toBe(5);
+        expect(response.fixture_id).toBe(3);
+        expect(response.response).toBe("2");
+        expect(typeof response.created_at).toBe("string");
+      });
+  });
+
+  test("PUT 201: Should update an existing response object from responses and return it", () => {
```
Comment (miors) [inline back-end/__tests__/app.test.js]: These 2 tests are doing the same thing, perhaps remove one of them?
```
test("PUT 201: Should insert a response object into responses and return it"
test("PUT 201: Should update an existing response object from responses and return it
```

File: back-end/models/responses-models.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    .query(
+      `
+  DELETE FROM responses WHERE user_id = $1 AND fixture_id = $2;
+  `,
+      [user_id, fixture_id]
+    )
+    .then(() => {
+      return db.query(
+        `
+    INSERT INTO responses (user_id, fixture_id, response)
```
Comment (miors) [inline back-end/models/responses-models.js:15]: Is it necessary to delete and then insert. Can't we achieve the same thing by just using update command?

**Review by miors — CHANGES_REQUESTED** [review]
HI @HanzoDaBoss , great work so far 👍 . Only a few comments as below. Also, need to include PUT /api/responses in the endpoints.json file.

File: back-end/__tests__/app.test.js (hunk 2 above)
Comment (Stephen0wen) [reply back-end/__tests__/app.test.js]: They are doing slightly differnt things - the first one is for creating a new response in the db, while the second one is for updating an existing one.

File: back-end/models/responses-models.js (hunk 3 above)
Comment (Stephen0wen) [reply back-end/models/responses-models.js:15]: It needs to also work when the response does not already exist

File: back-end/app.js [hunk 4]
```diff
@@ -2,23 +2,22 @@ const express = require("express");
...
 const {
-  handleCustomErrors,
-  handlePsqlErrors,
-  handleServerErrors,
+    handleCustomErrors,
+    handlePsqlErrors,
+    handleServerErrors,
 } = require("./errors/index.js");
 const { getTeams } = require("./controllers/teams-controllers.js");
-const { getFixturesById } = require("./controllers/fixtures_byId.controller");
```
Comment (miors) [inline back-end/app.js]: Need to reinsert this back in

File: back-end/app.js [hunk 5]
```diff
@@ -39,14 +38,12 @@ app.get("/api/venues/:venue_id", getVenueById);
 
 app.get("/api/teams", getTeams);
 
-app.get("/api/fixtures/:fixture_id", getFixturesById);
```
Comment (miors) [inline back-end/app.js]: Need to reinsert this back in

File: back-end/app.js [hunk 6]
```diff
@@ -39,14 +38,12 @@ app.get("/api/venues/:venue_id", getVenueById);
...
-
 app.get("/api/fixtures", getFixtures);
 
-app.delete("/api/users/:user_id", deleteUserByUserId);
```
Comment (miors) [inline back-end/app.js]: Need to reinsert this back in

**Review by miors — CHANGES_REQUESTED** [review]
Hi @Stephen0wen , some tests are failing because a few lines of code have been removed during conflict resolution 😅 . See comments below.

File: back-end/app.js [hunk 7]
```diff
@@ -2,23 +2,22 @@ const express = require("express");
 const { getApi } = require("./controllers/api.controller");
 
 const {
-  getUsers,
-  postUser,
-  patchUserById,
-  getUserById,
-  deleteUserByUserId,
```
Comment (miors) [inline back-end/app.js]: Need to reinsert this back in

**Review by HanzoDaBoss — COMMENTED** [review]
It's resolved

**Review by miors — APPROVED** [review]
LGTM. Great work @HanzoDaBoss ans @Stephen0wen 👍

**Pull request merged**